avm/src/daemon/ctrl.rs

```diff
@@ -400,4 +586,169 @@ impl ControlPort {
       }
     }
   }
+
+  pub fn get_vm_for_key(self: &ControlPort, key: &str) -> &VMMetadata {
+    &self.vms[self.lrh.get_primary_node_id(key)]
+  }
+
+  pub fn is_key_owner(self: &ControlPort, key: &str) -> bool {
+    match &self.self_vm {
+      Some(my) => self.get_vm_for_key(key).private_ip_addr == my.private_ip_addr,
+      None => false,
+    }
+  }
+
+  pub async fn dsgetf(self: &ControlPort, key: &str) -> Option<Arc<HandlerMemory>> {
+    match self.dsgetf_inner(key).await {
+      Ok(hm) => Some(hm),
+      Err(_) => None,
+    }
+  }
+
+  async fn dsgetf_inner(self: &ControlPort, key: &str) -> DaemonResult<Arc<HandlerMemory>> {
+    let vm = self.get_vm_for_key(key);
+    let url = format!("https://{}:4142/datastore/getf", vm.public_ip_addr);
+    let req = Request::builder().method("POST").uri(url);
+    let cluster_secret = CLUSTER_SECRET.get().unwrap().as_ref().unwrap().clone();
```

Comment:
2 nits:
1. should we handle the error case?
2. could be `.get().unwrap().clone().unwrap();`

Reply:
1. The `daemon` refuses to spin up without a cluster secret. And this path can only be called in `daemon` mode because the control port is not created otherwise.
2. I'll try it and see.

Reply:
i forgot this was in the daemon for a second